examples/skr/TROUBLESHOOTING.md

```diff
@@ -53,3 +53,57 @@ err: AKV post request failed: HTTP GET failed: Post "https://<mhsm-name>.managed
+
+Issue the following commands to check the workload identity is properly enabled in the namespace where skr container pod is run: 
+
+```bash
+# Obtain the client id of the managed identity. 
+# USER_ASSIGNED_IDENTITY_NAME is the name of the managed identity used for accessing key vault and RESOURCE_GROUP is the resource group name where the managed identity resides 
+export USER_ASSIGNED_CLIENT_ID="$(az identity show --resource-group "${RESOURCE_GROUP}" --name "${USER_ASSIGNED_IDENTITY_NAME}" --query 'clientId' -otsv)" 
+# Obtain the service account detail
+kubectl get sa <service-account-name> -n <skr-container-namespace> -o yaml 
+```
+
+Check the service account `azure.workload.identity/client-id` annotation value matches the obtained `USER_ASSIGNED_CLIENT_ID` value. If they match, check the following: 
+
+```bash 
+# Obtain the AKS_OIDC_ISSUER. Replace CLUSTER_NAME and RESOURCE_GROUP with the name of the cluster skr is run and the resource group the cluster resides 
+export AKS_OIDC_ISSUER="$(az aks show -n "${CLUSTER_NAME}" -g "${RESOURCE_GROUP}" --query "oidcIssuerProfile.issuerUrl" -otsv)"
+# Obtain the federated identity detail 
+# USER_ASSIGNED_IDENTITY_NAME is the name of the managed identity used for accessing key vault
+# RESOURCE_GROUP is the resource group name where managed identity resides 
+# FEDERATED_IDENTITY_NAME is the name of the federated identity created previously before skr is initially run 
+az identity federated-credential show --resource-group "${RESOURCE_GROUP}" -n "${FEDERATED_IDENTITY_NAME}" --identity-name "${USER_ASSIGNED_IDENTITY_NAME}"
+```
+
+Once you have the federated identity detail, it should look something like this: 
+
+```json
+{
+  "audiences": [
+    "api://AzureADTokenExchange"
+  ],
+  "id": "/subscriptions/84**********e9bdf8/resourcegroups/resourcegroupname/providers/Microsoft.ManagedIdentity/userAssignedIdentities/identityname/federatedIdentityCredentials/myFedIdentity",
+  "issuer": "https://eastus.oic.prod-aks.azure.com/72f988**********011db47/ddfa**********2d5/",
+  "name": "myFedIdentity",
+  "resourceGroup": "resourcegroupname",
+  "subject": "system:serviceaccount:kafka:workload-identity-sa",
+  "systemData": null,
+  "type": "Microsoft.ManagedIdentity/userAssignedIdentities/federatedIdentityCredentials"
+}
+```
+
+Make sure the issuer matches the `AKS_OIDC_ISSUER` value obtained, the subject matches the service account especially the namespace and service account name part, the resourceGroup matches the one managed identity resides. If any one of the above items does not match, please follow the guide to re-enable workload identity. After re-enabling the workload identity and it still does not solve the issue and users are still seeing `CreateContainerRequest` error, then we can safely rule out that the error does not come from workload identity setup. 
```

Comment:
"Make sure the issuer matches the `AKS_OIDC_ISSUER` value obtained, the subject matches the service account (especially the namespace and service account name part), and the resourceGroup matches the one where the managed identity resides."